### ISSUE TITLE
Converted solution to VS2017 format and dotnet test, and now missing coverage

### ISSUE THREAD

**Issue by snakefoot** [OpenCover/opencover#756]
### My Framework

- [x] .NET 4.5
- [x] .NET 4.6.1

### My Environment

- [x] AppVeyor (VS2017 Image - OpenCover.4.6.519)

### I have already...

- [x] repeated the problem using the latest stable release of OpenCover.
- [x] reviewed the [usage guide](https://github.com/opencover/opencover/wiki/usage) and [usage document](https://github.com/opencover/opencover/blob/master/main/OpenCover.Documentation/Usage.pdf).
- [x] have looked at the opencover output xml file in an attempt to resolve the issue.

### My issue is related to (check only those which apply):

- [x] almost no coverage being recorded

### Expected Behavior

Should include coverage of entire library-code.

### Actual Behavior

When using filter, then it only include a single class (BaseMutexFileAppender.cs)

Without filter then it shows the correct coverage of the unit-test-library, but still only a single class of the library being tested.

### Steps to reproduce the problem:

- Execute OpenCover with dotnet test on AppVeyor VS2017-Image:

`OpenCover.Console.exe -register:user -target:"C:\Program Files\dotnet\dotnet.exe" -targetargs:"test tests\NLog.UnitTests --configuration Debug --framework net461" -returntargetcode -filter:"+[NLog]* +[NLog.Extended]* -[NLog]JetBrains.Annotations.*" -excludebyattribute:*.ExcludeFromCodeCoverage* -hideskipped:All -output:coverage.xml`

- [Broken coverage result](https://codecov.io/gh/NLog/NLog/pull/2263?src=pr&el=desc) - Notice how it only shows for BaseMutexFileAppender.cs
- [Broken coverage build report](https://codecov.io/gh/NLog/NLog/commit/285726bd536b39a63d0db38e4b23acc8d773d381/build)
- [appveyor.yml](https://github.com/snakefoot/NLog/blob/285726bd536b39a63d0db38e4b23acc8d773d381/appveyor.yml)
- [Library csproj](https://github.com/snakefoot/NLog/blob/285726bd536b39a63d0db38e4b23acc8d773d381/src/NLog/NLog.csproj)
- [UnitTest csproj](https://github.com/snakefoot/NLog/blob/285726bd536b39a63d0db38e4b23acc8d773d381/tests/NLog.UnitTests/NLog.UnitTests.csproj)

**Comment by sawilde** [comment]
You'll probably need to try that latest beta build to get coverage from anything built using `dotnet` this is due to changes in pdb formats that are not supported by the current release.

https://ci.appveyor.com/project/sawilde/opencover/build/4.6.780/artifacts

**Comment by snakefoot** [comment]
@sawilde Funny how the UnitTest-classes gets coverage, but not the actually Library-classes.

Possible for you to publish a release-candidate nuget-package, that I can grap from the NuGet-feed (easier to use on appveyor.yml) ?

**Comment by sawilde** [comment]
@snakefoot not yet as it isn't ready - publishing a release candidate creates an expectation

**Comment by sawilde** [comment]
There is a myget feed - https://www.myget.org/feed/Packages/opencover

**Comment by snakefoot** [comment]
@sawilde Is this a public feed ? I cannot see to access it.

**Comment by sawilde** [comment]
@snakefoot I believe so but you may need a myget account perhaps.

try this:

```
nuget.exe install OpenCover -Version 4.6.780-rc -Source https://www.myget.org/F/opencover/api/v3/index.json
```

**Comment by snakefoot** [comment]
@sawilde Tried the BETA-build, but it is not collecting coverage either:

https://codecov.io/gh/NLog/NLog/commit/615e6d61353a230968c87d5c4dc3641a7f8ca29c/build

**Comment by snakefoot** [comment]
Discovered that xUnit 2.0 xunit.console.x86.exe is still able to execute the tests, and now it the result is this:

```
=== TEST EXECUTION SUMMARY ===
   NLog.UnitTests  Total: 2223, Errors: 0, Failed: 0, Skipped: 6, Time: 412,726s
Committing...
Visited Classes 390 of 418 (93.3)
Visited Methods 3241 of 3695 (87.71)
Visited Points 19233 of 22821 (84.28)
Visited Branches 7379 of 9068 (81.37)
```

**Comment by 304NotModified** [comment]
Have the same problem here (https://github.com/NLog/NLog.MailKit/pull/5), and that's a lot smaller repo (so easier to test)

**Comment by quetzalcoatl** [comment]
I've noticed @snakefoot was effectively using `dotnet.exe test ... --framework net461` as the OpenCover target command. I have just succeeded in running similar thing.

I work on Win10-Home x64, VS2017 Comunity, OpenCover 4.6.519, xunit 2.2.0 and 2.3.1 (no diffrence)

My story is - I have a `netcoreapp2.0` project that I wanted to cover. I succeeded, and although that's offtopic another story I've yet to blog about, let me summarize the most important parts for comparison:

(netcoreapp2.0) In covered project csproj:

    <TargetFramework>netcoreapp2.0</TargetFramework>
    <DebugType>full</DebugType>

(netcoreapp2.0) In test project:

    <TargetFramework>netcoreapp2.0</TargetFramework>
    

(netcoreapp2.0) OpenCover runner (single command line, multiline for readability, I've left only the important parts):

    OpenCover.Console.exe
        -register:user
        -target:"C:\Program Files\dotnet\dotnet.exe"
        -targetargs:"test --framework netcoreapp2.0 --logger:trx \"MyProject\MyProject.csproj\" "
        -coverbytest:*.Tests.dll
        -output:"MyProject\TestResults\results-coverage.xml"
        -oldstyle
        -log:Warn

I've spent few days (really, not exagerrating) on researching different options, and that's the only combination that worked. Similar setup worked for me for netcoreapp1.1 and netcoreapp2.0. Note that I had to enable `Full` PDBs only for "real projects", and enabling them for test projects did some damage, although I don't remember what exactly (sorry, that was over 2 months ago). Another thing to note is the `-oldstyle` switch already discussed in other issues. `Warn` or `Info` log level helps you to see "No PDBs" infos which help in diagnosing when you get empty reports. I also had to set up `output` and `logger` due to multiple projects and multiple test projects to cover in one go.

Now the fun part. I had to switch my net-core-app to net461 due to 3rd party libraries. Application works fine, tests run fine, opencover (`--framework net461`) returns:

    (log level: info)
    Cannot instrument [SNIP]\AppData\Local\Temp\[SNIP]\assembly\dl3\[SNIP]\Foobar.WebApi.exe as no PDB/MDB could be loaded

    (log level: warn)
    No results, this could be for a number of reasons. The most common reasons are:
        1) missing PDBs for the assemblies that match the filter please review the
        output file and refer to the Usage guide (Usage.rtf) about filters.
        2) the profiler may not be registered correctly, please refer to the Usage
        guide and the -register switch.

I similar messages when I battled with `netcoreapp`, resolved by using `full` PDBs. Now obviously it was still set, as I just switched the target framework. After another long research, I found out that ... it actually can not locate the PDBs for some reason. Adding `-searchdirs` parameter made it work for `net461`! Also, `oldstyle` is no longer needed as runtime is 461 and new instrumentation can be used. For my projects, I left the oldstyle and also left the full PDBs, maybe I'll be switching back to netcore at some point.

(net461) In covered project csproj:

    <TargetFramework>net461</TargetFramework>
    <DebugType>full</DebugType>

(net461) In test project:

    <TargetFramework>net461</TargetFramework>
    

(net461) OpenCover runner (single command line, multiline for readability, I've left only the important parts):

    OpenCover.Console.exe
        -register:user
        -target:"C:\Program Files\dotnet\dotnet.exe"
        -targetargs:"test --framework net461 --logger:trx \"MyProject\MyProject.csproj\" "
        -coverbytest:*.Tests.dll
        -output:"MyProject\TestResults\results-coverage.xml"
        -searchdirs:"MyProject\bin\Debug\net461"
        -log:Warn

I can only guess that either opencover does not look in \bin\$config\$platform for them, or dotnet test/vstest does not copy them properly, or xunit-runner does not copy/shadow them properly..

Anyways I'm absolutely hyped. For the 3rd time in a row (core 1.1, core 2.0, net461), I was dead-tired trying to get it working and almost gave up, and yet again it works out of the box with OpenCover 4.6.519 provided you somehow happen to guess the switches. Unfortunatelly, I cannot share the full project setup, and can't prepare a new self-contained project demos (time/effort/nda/license/etc), I'll try to do that a bit later.
